include/cjdb/contracts.hpp

```diff
@@ -4,10 +4,42 @@
 #ifndef CJDB_CONTRACTS_HPP
 #define CJDB_CONTRACTS_HPP
 
-#include <cstdio>
 #include <string_view>
 #include <type_traits>
 
+#ifdef _MSC_VER
+	#define CJDB_PRETTY_FUNCTION __FUNCSIG__
+	#define CJDB_FORCE_INLINE __forceinline
+#else
+	#define CJDB_PRETTY_FUNCTION __PRETTY_FUNCTION__
+	#define CJDB_FORCE_INLINE [[gnu::always_inline]] inline
+#endif // _MSC_VER
+
+#ifndef CJDB_PRINT_ERROR
+	#ifdef CJDB_USE_IOSTREAM
+		#include <iostream>
+	#else
+		#include <cstdio>
+	#endif // CJDB_USE_IOSTREAM
+
+	namespace cjdb::contracts_detail {
+		struct print_error_fn {
+			CJDB_FORCE_INLINE void operator()(std::string_view const message) const noexcept
+	#ifdef CJDB_USE_IOSTREAM
+			try {
+				std::clog.write(message.data(), static_cast<std::streamsize>(message.size()));
+			} catch(...) {}
```

Comment:
try/catch block isn't necessary: if the stream throws in the middle of reporting a contract violation, I think it would be more useful if the user discovered that.

Reply:
Ok, I got rid of that and for consistency made the stdio implementation throw when `fwrite` fails.